hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/ShowTimelineProcedure.scala

```diff
@@ -0,0 +1,543 @@
+    val metaClient = createMetaClient(jsc, basePath)
+
+    val timelineEntries = getTimelineEntries(metaClient, limit, showArchived, startTime, endTime)
+
+    applyFilter(timelineEntries, filter, outputType)
+  }
+
+  override def build: Procedure = new ShowTimelineProcedure()
+
+  /**
+   * Retrieves timeline entries from both active and archived timelines based on the provided parameters.
+   *
+   * This is the main orchestration method that coordinates the retrieval of timeline entries. It:
+   * 1. Builds instant information map from the active timeline (for modification time lookups)
+   * 2. Loads rollback information for both active and archived timelines
+   * 3. Optionally loads archived timeline details if `showArchived` is true
+   * 4. Retrieves entries from active timeline
+   * 5. Optionally retrieves entries from archived timeline and combines them
+   * 6. Sorts combined entries by timestamp (descending) and state priority
+   * 7. Applies limit or time range filtering as appropriate
+   *
+   * @param metaClient The Hudi table metadata client for accessing timeline data
+   * @param limit Maximum number of entries to return (ignored if time range is specified)
+   * @param showArchived Whether to include archived timeline entries in the result
+   * @param startTime Optional start timestamp for filtering (format: yyyyMMddHHmmss). If empty, no start filtering is applied
+   * @param endTime Optional end timestamp for filtering (format: yyyyMMddHHmmss). If empty, no end filtering is applied
+   * @return Sequence of Row objects representing timeline entries, sorted by timestamp (descending) and state priority
+   *         (COMPLETED > INFLIGHT > REQUESTED)
+   *
+   * @note When `showArchived` is true, this method will:
+   *       - Reload the archived timeline to ensure it's up-to-date
+   *       - Load completed instant details and compaction details into memory for the specified time range or limit
+   *       - Combine active and archived entries, with archived entries marked with "ARCHIVED" timeline type
+   *
+   * @note The sorting logic prioritizes:
+   *       1. Timestamp (newer first)
+   *       2. State (COMPLETED > INFLIGHT > REQUESTED)
+   *
+   * @see [[getTimelineEntriesFromTimeline]] for extracting entries from a specific timeline
+   * @see [[buildInstantInfoFromTimeline]] for building the instant information map
+   */
+  private def getTimelineEntries(metaClient: HoodieTableMetaClient,
+                                 limit: Int,
+                                 showArchived: Boolean,
+                                 startTime: String,
+                                 endTime: String): Seq[Row] = {
+
+    val instantInfoMap = buildInstantInfoFromTimeline(metaClient)
+
+    val activeRollbackInfoMap = getRolledBackInstantInfo(metaClient.getActiveTimeline, metaClient)
+    val archivedRollbackInfoMap = if (showArchived) {
+      val archivedTimeline = metaClient.getArchivedTimeline.reload()
```

Comment:
this line will triggers the whole timeline instants loading twice already, `metaClient.getArchivedTimeline` for one time with load mode as `LoadMode.ACTION`, and `.reload()` another time with load mode as `LoadMode.ACTION`, note that for V2, only the `FULL` mode supports reading the plans(rollback, cleaning and compaciton).

Reply:
maybe we just new an empty timeline there and invokes these APIs.